File: 70-climbing-stairs/climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2: return n
        dp_1, dp_2 = 1, 2
        sol = 0
        for i in range(3, n+1):
            sol = dp_1 + dp_2
            dp_1 = dp_2
            dp_2 = sol

        return sol
            
        

File: 70-climbing-stairs/climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n <= 2){
            return n;
        }
        int x = 1, y = 2, sol = 0;

        for (int i=3; i<n+1; i++){
            sol  = x + y;
            x = y, y = sol;
        }

        return sol;
        
    }
};